x/posts/internal/types/post.go

```diff
@@ -43,12 +42,11 @@ func ParsePostId(value string) (PostId, error) {
 
 // Post is a struct of a Magpie post
 type Post struct {
-	PostID        PostId         `json:"id"`
-	ParentID      PostId         `json:"parent_id"`
+	PostID        PostID         `json:"id"`
+	ParentID      PostID         `json:"parent_id"`
 	Message       string         `json:"message"`
-	Created       time.Time      `json:"created"`
-	Modified      time.Time      `json:"modified"`
-	Likes         uint           `json:"likes"`
+	Created       int64          `json:"created"`     // Block height at which the post has been created
+	LastEdited    int64          `json:"last_edited"` // Block height at which the post has been edited the last time
 	Owner         sdk.AccAddress `json:"owner"`
```

Comment:
The `Likes` is deleted here. It was there to have a quick reference on how many likes have been casted to the post. If it doesn't exist, would it be easy to get this number? I doubt. LevelDB is simple key value pair data structure and seems these aggregation feature doesn't exist?

Reply:
The problem is that currently we are storing likes in the wrong way. 
Right now when a Like object is saved inside the local database, it is saved using it's id as the key as the whole object itself as the value.  

I personally think we should store the different likes for a post using the post id as the key and the array of likes as the value.   
In this way we could easily retrieve the array of likes that a post has associated to itself (having the id of the post) and count how many likes the post has. This could be done inside the keeper's `GetPost` method or, even better, inside the `querier` method that handles the reading of posts or a specific post's details. 

Also, with this new method of storing likes when a user wants to unlike a post we will have to go through just the likes objects associated with that post and find the one for that user.  
The same would happen when a user tries to like a post and we need to make sure he hasn't liked it yet. 
Right now, those operations are performed in `O(n)` where `n` is the overall number of likes to all the posts. With the new method they could be performed in `O(m)` with `m` being the number of the likes to the specific post. Since we have `m <= n` for each `n` chosen, I think we should change how we store the likes in that way. 

Even when retrieving the number of likes per post we can do it in `O(1)` as we have the post id and can easily get the array of likes, so even in this case we do not have any particular problem.

Reply:
Make sense. You are right. Actually I did the same way in a previous webapp using MongoDB before. I have overlooked this issue here.